modules/js-and-css/speculation-rules/settings.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Settings functions used for Speculation Rules.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Returns the available options for the Speculation Rules mode and their labels.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Associative array of `$mode => $label` pairs.
+ */
+function plsr_get_mode_labels() {
+	return array(
+		'prefetch'  => _x( 'Prefetch', 'setting label', 'performance-lab' ),
+		'prerender' => _x( 'Prerender', 'setting label', 'performance-lab' ),
+	);
+}
+
+/**
+ * Returns the available options for the Speculation Rules eagerness and their labels.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Associative array of `$eagerness => $label` pairs.
+ */
+function plsr_get_eagerness_labels() {
+	return array(
+		'conservative' => _x( 'Conservative (on click)', 'setting label', 'performance-lab' ),
+		'moderate'     => _x( 'Moderate (on hover)', 'setting label', 'performance-lab' ),
+		'eager'        => _x( 'Eager (on slightest suggestion)', 'setting label', 'performance-lab' ),
```

Comment:
BTW, at the last moment we split this into `"eager"` and `"immediate"`. Currently they're synonyms in Chrome, but the expectation is that `"immediate"` will continue to mean "right away, like on page load" and `"eager"` will be like this describes, "on the slightest suggestion".

This is probably still fine ("all links immediately" is probably not a very useful setting for a WP site), just mentioning it for context.

Reply:
I saw the `immediate` in the spec, but when reading came to the same conclusion, that it would seem too excessive to be realistically of value in the WordPress context - a bit too "aggressive" :)